.ci/build_demo.sh

```diff
@@ -0,0 +1,150 @@
+
+# Fetch all domain IDs
+DOMAIN_IDS=$(cat "$WORKSPACE/app/config/domains_urls.yml" | grep -Po 'id: ([0-9]+)$' | sed -r 's/id: ([0-9]+)/\1/')
+
+# Modify public URLs to $DOMAIN_ID.$DEMO_NAME.$DEMO_DOMAIN ($DOMAIN_ID is omitted for the first domain)
+for DOMAIN_ID in $DOMAIN_IDS; do
+    if [[ "$DOMAIN_ID" == "1" ]]; then
+        # 1st domain has URL without a number prefix
+        sed -i "/id: 1/,/url:/{s/url:.*/url: http:\/\/$DEMO_NAME.$DEMO_DOMAIN/}" "$WORKSPACE/app/config/domains_urls.yml"
+    else
+        # 2nd and subsequent domains have URLs with DOMAIN_ID prefix
+        sed -i "/id: $DOMAIN_ID/,/url:/{s/url:.*/url: http:\/\/$DOMAIN_ID.$DEMO_NAME.$DEMO_DOMAIN/}" "$WORKSPACE/app/config/domains_urls.yml"
+    fi
+done
+
+echo -e "${BLUE}Connecting to shared elasticsearch instance...${NC}"
+
+NETWORK_CONNECT='false'
+
+if docker network create "$ELASTICSEARCH_NETWORK" &> /dev/null; then
+    echo "A network $ELASTICSEARCH_NETWORK has been created."
+    NETWORK_CONNECT='true'
+else
+    echo "The network $ELASTICSEARCH_NETWORK already exists."
+fi
+
+if [[ -z "$(docker ps -q -f name="$ELASTICSEARCH_CONTAINER")" ]]; then
+    echo "Creating a shared elasticsearch container $ELASTICSEARCH_CONTAINER..."
+    docker run --name "$ELASTICSEARCH_CONTAINER" --ulimit "nofile=65536:65536" --env "discovery.type=single-node" -d "docker.elastic.co/elasticsearch/elasticsearch-oss:6.3.2"
+    echo "The elasticsearch container has been created."
+    NETWORK_CONNECT='true'
+else
+    echo "The elasticsearch container $ELASTICSEARCH_CONTAINER already exists."
+fi
+
+if [[ "$NETWORK_CONNECT" == 'true' ]]; then
+    echo "Connecting the container to the network...."
+    docker network connect "$ELASTICSEARCH_NETWORK" "$ELASTICSEARCH_CONTAINER"
+fi
+
+ELASTICSEARCH_IP=$(docker inspect -f "{{ with (index .NetworkSettings.Networks \"$ELASTICSEARCH_NETWORK\") }}{{ .IPAddress }}{{ end }}" "$ELASTICSEARCH_CONTAINER")
+if [[ -z "$ELASTICSEARCH_IP" ]]; then
+    echo -e "${RED}The IP address of the elasticsearch container inside the shared network could not be obtained!${NC}"
+    exit 1
+else
+    echo -e "The IP address of ${BLUE}$ELASTICSEARCH_CONTAINER${NC} inside ${BLUE}$ELASTICSEARCH_NETWORK${NC} is ${BLUE}$ELASTICSEARCH_IP${NC}"
+fi
+
+# Copy the demo Docker configuration
+cp docker/conf/docker-compose.demo.yml.dist docker-compose.yml
+
+echo -e "${BLUE}Building Docker images and starting up the containers...${NC}"
+
+# Export variables used in Docker Compose configuration
+export DEMO_NAME
+export ELASTICSEARCH_NETWORK
+export ELASTICSEARCH_IP
+
+# Build the Docker image and start the containers
+docker-compose up --build --force-recreate -d
+
+# Configure the nginx
+WEBSERVER_PORT=$(docker-compose port webserver 8080)
+echo "upstream $JOB_NAME-upstream {
+    server 127.0.0.1:${WEBSERVER_PORT##*:};
+}
+
+server {
+    listen 80;
+    server_name $DEMO_NAME.$DEMO_DOMAIN *.$DEMO_NAME.$DEMO_DOMAIN;
```

Comment:
For it to be URL, the protocol would have to be included, so it would still be just a domain. Then a second URL (for second domain) would have to be created. The DEMO_NAME is also used in docker-compose.yml as a prefix in elasticsearch.

I guess it could be rewritten so it just works with DEMO_DOMAIN variable, but I don't see the need atm.

Reply:
You can set the DEMO_NAME variable from $JOB_NAME and you can still join these two variables into one